Update dead link to removed "Horse saddle" item

rybak opened the pull request:
Minecraft Wiki article is in the process of being split. The "Horse
saddle" section is no longer part of the page "Java Edition removed
features", but is located at "Java Edition removed items".

rybak commented:
Oops, wrong branch. Replaced with #17.

rybak commented:
#17 is closed because its destination branch is gone. So now this PR makes more sense.

Joshimuz commented:
I would merge this but there's a conflict (coz I updated the readme) and I can't figure out the resolve conflict tool :D

rybak committed:
Update dead link to removed "Horse saddle" item

Minecraft Wiki article is in the process of being split. The "Horse
saddle" section is no longer part of the page "Java Edition removed
features", but is located at "Java Edition removed items".

rybak commented:
IntelliJ IDEA's merge resolver tool did it automagically, which is nice.
